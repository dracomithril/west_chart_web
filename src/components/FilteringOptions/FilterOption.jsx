/**
 * Created by Gryzli on 23.04.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'react-bootstrap';

const FilterOption = ({ control, input, description }, { store }) => {
  const { filters } = store.getState();
  return (
    <Checkbox
      {...control}
      checked={filters[input.name].checked}
      onChange={({ target }) => {
        const { id, checked } = target;
        store.dispatch({ type: 'TOGGLE_FILTER', id, checked });
      }}
    >
      {description.start}
      <input
        className="num_days"
        type="number"
        min={0}
        step={1}
        id={control.id}
        value={filters[input.name].days}
        onChange={({ target }) => {
          const { id, name, value } = target;
          store.dispatch({
            type: 'UPDATE_DAYS',
            id,
            name,
            value: Number(value),
          });
        }}
      />
      {description.end}
    </Checkbox>
  );
};
FilterOption.contextTypes = {
  store: PropTypes.object,
};
FilterOption.propTypes = {
  control: PropTypes.object,
  input: PropTypes.object,
  description: PropTypes.object,
};
export default FilterOption;
