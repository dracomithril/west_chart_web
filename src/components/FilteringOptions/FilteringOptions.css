.panel {
    padding: 0;
    margin: 0 0 5px;
    /*position: relative;*/
    /*z-index: 1000;*/
}

.panel-heading {
    padding: 5px 5px;
    border-bottom: 1px solid transparent;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
}

.panel-body {
    padding: 5px;
    text-align: left;
}

.num_days {
    padding-left: 5px;
    text-align: right;
    width: 50px;
}

.filter_panel {
    float: left;
    border: 1px solid #bce8f1;
    /*background: azure;*/
    /*margin-left: 10px;*/
    text-align: left;
    padding-left: 10px;
    padding-right: 5px;
    height: 100px;
    font-size: smaller;
    /*overflow-y: scroll;*/
    border-radius: 3px;
}
