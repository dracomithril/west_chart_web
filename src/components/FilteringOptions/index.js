export { default, componentWithStyles } from './FilteringOptions';
