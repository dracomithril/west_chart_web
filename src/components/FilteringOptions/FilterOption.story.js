import React from 'react';
import { addDecorator, storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Provider } from 'react-redux';
import FilterOption from './FilterOption';
import configureStore from '../../configureStore';

const state = {
  filters: {
    woc_control: {
      checked: true,
    },
  },
};
const store = configureStore(state);

addDecorator(story => <Provider store={store}>{story()}</Provider>);
storiesOf('FilterOption', module)
  .add('Todo[VR]', () => <FilterOption />)
  .add('Checked[VR]', () => (
    <FilterOption
      checked
      onChange={action('checked')}
      desc_start="some filter"
      desc_end="days"
      onValueChange={action('value changed')}
    />
  ));
