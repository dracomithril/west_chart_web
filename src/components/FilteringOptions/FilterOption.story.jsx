import React from 'react';
import { addDecorator, storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Provider } from 'react-redux';
import FilterOption from './FilterOption';
import configureStore from '../../configureStore';

const state = {
  filters: {
    woc_control: {
      checked: true,
    },
  },
};
const store = configureStore(state);

addDecorator(story => (
  <Provider store={store}>
    {story()}
  </Provider>
));
storiesOf('FilterOption', module)
  .add('Todo[VR]', () => <FilterOption />)
  .add('Checked[VR]', () => (
    <FilterOption
      checked
      onChange={action('checked')}
      descStart="some filter"
      descEnd="days"
      onValueChange={action('value changed')}
    />
  ));
