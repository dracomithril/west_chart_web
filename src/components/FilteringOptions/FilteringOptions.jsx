/**
 * Created by Gryzli on 28.01.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import FilterOption from './FilterOption';
import './FilteringOptions.css';
import filters_def from '../../utils/filters_def';
import { actionTypes } from '../../reducers/actionTypes';
import MessageControl from './MessageControl';

const styles = theme => ({
  button: {
    alignSelf: 'stretch',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    fontWeight: theme.typography.fontWeightRegular,
  },
  root: {
    width: '100%',
  },
});

const FilteringOptions = ({ classes }, { store }) => {
  const { filters } = store.getState();
  const map_c = filters_def.control.map(({ input, description, control }) => {
    const { days, checked } = filters[input.name] || {};
    return (
      <FilterOption
        id={control.id}
        days={days}
        name={control.name}
        checked={checked}
        descStart={description.start}
        descEnd={description.end}
        key={input.name}
        onChange={target => {
          store.dispatch({
            type: actionTypes.TOGGLE_FILTER,
            id: target.value,
            checked: target.checked,
          });
        }}
        onValueChange={target => {
          const { id, name, value } = target;
          store.dispatch({
            type: actionTypes.UPDATE_DAYS,
            id,
            name,
            value: Number(value),
          });
        }}
      />
    );
  });
  const map_t = filters_def.text.map(({ control, text, input }) => (
    <MessageControl
      name={control.name}
      text={text}
      id={control.id}
      checked={(filters[input.name] || {}).checked}
      key={input.name}
      onChange={target => {
        store.dispatch({
          type: actionTypes.TOGGLE_FILTER,
          id: target.value,
          checked: target.checked,
        });
      }}
    />
  ));
  return (
    <ExpansionPanel className={classes.root}>
      <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
        <Typography className={classes.heading}>Filters</Typography>
      </ExpansionPanelSummary>
      <ExpansionPanelDetails>
        <div>
          {map_c}
          {map_t}
        </div>
      </ExpansionPanelDetails>
    </ExpansionPanel>
  );
};

FilteringOptions.contextTypes = {
  store: PropTypes.shape,
};
FilteringOptions.propTypes = {
  classes: PropTypes.shape({
    button: PropTypes.string,
  }).isRequired,
};

export default withStyles(styles)(FilteringOptions);
