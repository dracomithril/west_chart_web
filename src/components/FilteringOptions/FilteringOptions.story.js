import React from 'react';
import { addDecorator, storiesOf } from '@storybook/react';
import { Provider } from 'react-redux';
import FilteringOptions from './index';
import configureStore from '../../configureStore';

const state = {
  filters: {
    woc_control: {
      checked: true,
    },
  },
};
const store = configureStore(state);

addDecorator(story => <Provider store={store}>{story()}</Provider>);
storiesOf('FilteringOptions', module).add('Todo[VR]', () => <FilteringOptions />);
