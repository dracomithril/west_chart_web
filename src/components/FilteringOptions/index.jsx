/**
 * Created by Gryzli on 28.01.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox, OverlayTrigger, Tooltip } from 'react-bootstrap';
import FilterOption from './FilterOption';
import './FilteringOptions.css';
import filters_def from '../../utils/filters_def';
import action_types from './../../reducers/action_types';

const _ = require('lodash');

const MessageControl = ({ input: { name }, text, control }, { store }) => {
  const { filters } = store.getState();
  return (
    <Checkbox
      checked={!filters[name].checked}
      onChange={({ target }) => {
        const { id, checked } = target;
        store.dispatch({ type: action_types.TOGGLE_FILTER, id, checked: !checked });
      }}
      {...control}
    >
      <OverlayTrigger
        placement="top"
        overlay={<Tooltip id={`${name}_tp`}>{`Will show all [${text}]`}</Tooltip>}
      >
        <span>[{text}]</span>
      </OverlayTrigger>
    </Checkbox>
  );
};

MessageControl.contextTypes = {
  store: PropTypes.object,
};
MessageControl.propTypes = {
  input: PropTypes.object,
  text: PropTypes.string,
  control: PropTypes.object,
};

const FilteringOptions = () => {
  const map_c = filters_def.control.map(elem => <FilterOption {...elem} key={elem.input.name} />);
  const map_t = filters_def.text.map(elem => <MessageControl {...elem} key={elem.input.name} />);
  const combined_map = [...map_t, ...map_c];
  const chunk_combined_map = _.chunk(combined_map, Math.floor(combined_map.length / 2));
  return (
    <div style={{ display: '-webkit-box' }}>
      <div className="filter_panel">{chunk_combined_map[0]}</div>
      <div className="filter_panel">{chunk_combined_map[1]}</div>
    </div>
  );
};
export default FilteringOptions;
