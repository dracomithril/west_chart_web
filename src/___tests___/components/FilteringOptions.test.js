/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import FilteringOptions from "./../../components/FilteringOptions/index";
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from "enzyme-to-json";
import configureMockStore from "redux-mock-store";

Enzyme.configure({ adapter: new Adapter() });
const initial_state = require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<FilteringOptions/>', () => {
  it('renders without crashing ChartPresenter', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <FilteringOptions/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  xit('should be able to change state of woc checkbox', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <FilteringOptions/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    let btn = wrapper.findWhere(n => n.props().id === 'bFilters').first();
    btn.simulate('click');
    let findWhere = wrapper.findWhere(n => n.props().id === 'woc_cb');
    const wocCheckbox = findWhere.first();
    let woc_props = wocCheckbox.props();
    expect(woc_props.checked).toBeTruthy();
    wocCheckbox.simulate('change', { target: { id: 'woc', checked: false } });
    let actions = store.getActions();
    expect(actions[0].type).toBe('TOGGLE_FILTER');
    expect(actions[0].id).toBe('woc');
    expect(actions[0].checked).toBeFalsy();
  })

});
