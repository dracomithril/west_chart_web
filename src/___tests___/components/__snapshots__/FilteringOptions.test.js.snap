// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilteringOptions/> renders without crashing ChartPresenter 1`] = `
<div
  style={
    Object {
      "display": "-webkit-box",
    }
  }
>
  <div
    className="filter_panel"
  >
    <MessageControl
      check={[Function]}
      control={
        Object {
          "id": "woc_cb",
          "name": "woc_text_control",
        }
      }
      input={
        Object {
          "name": "woc_control",
        }
      }
      key="woc_control"
      text="wielkie ogarnianie charta"
    />
    <MessageControl
      check={[Function]}
      control={
        Object {
          "id": "westletter_cb",
          "name": "westletter_text_control",
        }
      }
      input={
        Object {
          "name": "westletter_control",
        }
      }
      key="westletter_control"
      text="wcs weekly westletter"
    />
    <FilterOption
      check={[Function]}
      control={
        Object {
          "id": "create",
          "name": "date_create_control",
        }
      }
      description={
        Object {
          "end": " days",
          "start": "created in last ",
        }
      }
      input={
        Object {
          "max": 31,
          "name": "create_control",
        }
      }
      key="create_control"
      type="countDays"
      valueName="created_time"
    />
  </div>
  <div
    className="filter_panel"
  >
    <FilterOption
      check={[Function]}
      control={
        Object {
          "id": "update",
          "name": "date_update_control",
        }
      }
      description={
        Object {
          "end": " days",
          "start": "updated in last ",
        }
      }
      input={
        Object {
          "max": 31,
          "name": "update_control",
        }
      }
      key="update_control"
      type="countDays"
      valueName="updated_time"
    />
    <FilterOption
      check={[Function]}
      control={
        Object {
          "id": "more",
          "name": "more_then_control",
        }
      }
      description={
        Object {
          "start": "more then ",
        }
      }
      input={
        Object {
          "name": "more_control",
        }
      }
      key="more_control"
      type="more"
    />
    <FilterOption
      check={[Function]}
      control={
        Object {
          "id": "less",
          "name": "less_then_control",
        }
      }
      description={
        Object {
          "start": "less then ",
        }
      }
      input={
        Object {
          "name": "less_control",
        }
      }
      key="less_control"
      type="less"
    />
  </div>
</div>
`;

exports[`<FilteringOptions/> should be able to change state of woc checkbox 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      id="filters_p"
      placement="right"
    >
      <div>
        <Checkbox
          bsClass="checkbox"
          checked={true}
          disabled={false}
          id="woc"
          inline={false}
          onChange={[Function]}
        >
          <OverlayTrigger
            defaultOverlayShown={false}
            overlay={
              <Tooltip
                bsClass="tooltip"
                id="woc_tp"
                placement="right"
              >
                Will show all [Wielkie Ogarnianie Charta]
              </Tooltip>
            }
            placement="top"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <span>
              [
              Wielkie Ogarnianie Charta
              ]
            </span>
          </OverlayTrigger>
        </Checkbox>
        <span>
          Show only posts:
        </span>
        <br />
        <div
          style={
            Object {
              "paddingLeft": 10,
            }
          }
        >
          <FilterOption
            check={[Function]}
            control={
              Object {
                "id": "add",
                "name": "date_added_control",
              }
            }
            description={
              Object {
                "end": " days",
                "start": "added in last ",
              }
            }
            input={
              Object {
                "max": 31,
                "name": "add_control",
              }
            }
            type="countDays"
            valueName="added_time"
          />
          <FilterOption
            check={[Function]}
            control={
              Object {
                "id": "create",
                "name": "date_create_control",
              }
            }
            description={
              Object {
                "end": " days",
                "start": "created in last ",
              }
            }
            input={
              Object {
                "max": 31,
                "name": "create_control",
              }
            }
            type="countDays"
            valueName="created_time"
          />
          <FilterOption
            check={[Function]}
            control={
              Object {
                "id": "update",
                "name": "date_update_control",
              }
            }
            description={
              Object {
                "end": " days",
                "start": "updated in last ",
              }
            }
            input={
              Object {
                "max": 31,
                "name": "update_control",
              }
            }
            type="countDays"
            valueName="updated_time"
          />
          <FilterOption
            check={[Function]}
            control={
              Object {
                "id": "more",
                "name": "more_then_control",
              }
            }
            description={
              Object {
                "start": "more then ",
              }
            }
            input={
              Object {
                "name": "more_control",
              }
            }
            type="more_or_less"
          />
          <FilterOption
            check={[Function]}
            control={
              Object {
                "id": "less",
                "name": "less_then_control",
              }
            }
            description={
              Object {
                "start": "less then ",
              }
            }
            input={
              Object {
                "name": "less_control",
              }
            }
            type="more_or_less"
          />
        </div>
      </div>
    </Popover>
  }
  placement="bottom"
  rootClose={true}
  trigger={
    Array [
      "click",
    ]
  }
>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="info"
    disabled={false}
  >
    filters
  </Button>
</OverlayTrigger>
`;
